bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/GemsStrategy.scala

```diff
@@ -143,20 +144,21 @@ trait GemsStrategy extends AgsStrategy {
     // Create a set of the angles to try.
     val anglesToTry = (0 until 360 by 45).map(Angle.fromDegrees(_)).toSet
 
-    // A way to terminate the Mascot algorithm immediately in the following cases:
-    // 1. A usable 2 or 3-star asterism is found; or
-    // 2. If no asterisms were found.
-    // Returning false will stop the search
-    def progress(s: Strehl, usable: Boolean): Boolean = {
-      !((usable && s.stars.size >= 2) || (s.stars.size < 2))
-    }
+    // We can terminate the Mascot algorithm immediately if we find a usable 3-star asterism or no asterism is found.
+    def shouldContinue(s: Strehl, usable: Boolean): Boolean = !(
+      // A usable three star asterism.
+      (usable && s.stars.size >= 3) ||
+      // No asterism.
+      (s.stars.size < 2)
+    )
```

Comment:
I still get a bit confused.  Maybe if we don't flip the result of the comparison at the end as you suggested earlier?  I'm not sure what `usable` means but maybe:

````scala
def shouldContinue(s: Strehl, usable: Boolean): Boolean =
  !usable || s.tars.size == 2
````

Seems like we're saying that only 2 star asterisms require more work?